FWCore/Framework/src/Worker.h

```diff
@@ -836,17 +864,18 @@ namespace edm {
     template <>
     class CallImpl<OccurrenceTraits<ProcessBlockPrincipal, BranchActionProcessBlockInput>> {
     public:
-      typedef OccurrenceTraits<ProcessBlockPrincipal, BranchActionProcessBlockInput> Arg;
+      using Arg = OccurrenceTraits<ProcessBlockPrincipal, BranchActionProcessBlockInput>;
       static bool call(Worker* iWorker,
                        StreamID,
-                       ProcessBlockPrincipal const& pbp,
-                       EventSetupImpl const&,
+                       ProcessBlockTransitionInfo const& info,
                        ActivityRegistry* actReg,
                        ModuleCallingContext const* mcc,
                        Arg::Context const* context) {
         ModuleSignalSentry<Arg> cpp(actReg, context, mcc);
-        return iWorker->implDoAccessInputProcessBlock(pbp, mcc);
+        return iWorker->implDoAccessInputProcessBlock(info.principal(), mcc);
       }
+      static constexpr void esPrefetch(
+          Worker*, WaitingTask*, ServiceToken const&, ProcessBlockTransitionInfo const&, Transition) {}
```

Comment:
You need to check that the calls into here are not expecting this module to run the task if the task is at ref count 0.

Reply:
I think this is OK. This function is only called in one place. The ref count is incremented before the function is called and then decremented and checked after the function is called. Does that cover the case you are worried about?

```
    //Need to be sure the ref count isn't set to 0 immediately
    iTask->increment_ref_count();

    workerhelper::CallImpl<T>::esPrefetchAsync(this, iTask, token, transitionInfo, iTransition);
    edPrefetchAsync(iTask, token, principal);

    if (principal.branchType() == InEvent) {
      preActionBeforeRunEventAsync(iTask, moduleCallingContext_, principal);
    }

    if (0 == iTask->decrement_ref_count()) {
      //if everything finishes before we leave this routine, we need to launch the task
      tbb::task::spawn(*iTask);
    }
```

Reply:
agreed